### ISSUE TITLE
on_hull and on_clex_hull queries are misleading

### ISSUE THREAD

**Issue by bpuchala** [prisms-center/CASMcode#3]
The 'on_hull' and 'on_clex_hull' query results are misleading because they return 1 (true) for all qhull vertices, not just ground states. Additionally, qhull only keeps one point as a vertex when there are multiple points with approximately equal coordinates. Particularly with 'on_clex_hull', this means there may be cases where it returns 0 (false) for a configuration that is on the hull, but with the same energy and composition as another configuration. Because of this, most users have been using 'hull_dist' with a tolerance check.

Possible changes:
1) change `on_hull` to `is_groundstate` and have it evaluate `lt(abs(hull_dist(...args...)),hull_tol)`, and have a similar `is_clex_groundstate`. To reduce the number of arguments needed by `hull_dist` and `on_hull`, we could make the `dim_tol` and `bottom_tol` arguments project settings.
2) keep current functionality, but change the documentation
3) delete the `on_hull` and `on_clex_hull` options

**Comment by goirijo** [comment]
I see no reason to delete the functionality, but the current implementation doesn't really make sense. High energy configurations that are on the hull don't really mean anything important. I think the first option makes most sense, with a tolerance in the settings.

**Comment by shaliu19900911** [comment]
I found the same thing. But I don't understand where should I make the changes above?

**Comment by jcthomas** [comment]
I don't think we should delete the queries, but I am also opposed using `is_groundstate` and `is_clex_groundstate`. 'groundstate' implies (at least to me) a definitive statement about stability with respect to any configuration, while these queries only indicate energetic optimality within the specified dataset. 

My concern is that using 'groundstate' in a non-definitive context will cause naming confusion in the future for any queries that implement more robust determination of phase stability.

@bpuchala if you feel that the change in behavior should be accompanied by a change in naming, so as to avoid confusion, I would suggest `clex_hull_point` and `dft_hull_point`

**Comment by shaliu19900911** [comment]
I use 'casm query --help properties' but don't see 'is_groundstate'. Should I edit something?

**Comment by jcthomas** [comment]
@shaliu19900911 this is a discussion about adding a new query that fix the current problems with `on_hull` and `on_clex_hull`. For now you can use query aliases to add custom queries on a project-by-project basis. For example
````
casm query --alias "is_groundstate = lt(hull_dist(CALCULATED,0.00001))"
````
will allow you to use the query `is_groundstate` within your project as a shorthand for `lt(hull_dist(CALCULATED,0.00001))`.

**Comment by shaliu19900911** [comment]
Yes I know now. Thanks!
